src/utils/dao-errors.js

```diff
@@ -0,0 +1,38 @@
+/* eslint-disable max-classes-per-file */
+/* eslint-disable class-methods-use-this */
+
+class ResourceNotFoundError {
```

Comment:
I felt like it made more sense to define these errors as they came up, rather than trying to come up with an exhaustive list now.